### ISSUE TITLE
NODERAWFS bug in fseek/fwrite

### ISSUE THREAD

**Issue by JanWielemaker** [kripken/emscripten#6612]
Trying to figure out why `minizip` which is embedded in SWI-Prolog produces invalid zip files, I found that  `fseek()`/`fwrite()` do not function properly.  Below is a copy from the [discussion](https://github.com/SWI-Prolog/roadmap/issues/43)

```
#include <stdio.h>

int
main(int argc, char **argv)
{ FILE *f = fopen("test.data", "wb");
  char data[] = "0123456789abcdef";
  char data2[] = "0123456789ABCDEF";
  char zero[16] = {0};

  fwrite(data, 1, 16, f);
  fwrite(zero, 1, 16, f);
  fwrite(data, 1, 16, f);
  fseek(f, SEEK_SET, 16);
  fwrite(data2, 1, 16, f);

  fclose(f);

  return 0;
}
```

Run using

```
emcc -o t.bc t.c
emcc t.bc -s NODERAWFS=1 -o t.html
node t.js
od -c test.data 
0000000   0   1   2   3   4   5   6   7   8   9   a   b   c   d   e   f
0000020  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0
0000040   0   1   2   3   4   5   6   7   8   9   a   b   c   d   e   f
0000060   0   1   2   3   4   5   6   7   8   9   A   B   C   D   E   F
```

The zero block should have been written with data2, but instead, data2 is at the end.

Versions (on Ubuntu 18.04):

  - node v8.9.1
  - emcc 1.38.4

P.s.  The original code checks all return values.  These are fine.   I also inserted an ftell() call after the fseek() that confirms the file pointer moved, but it didn't.

**Comment by JanWielemaker** [comment]
Bypassing the FILE stream layer avoids the problem: this works fine:

```
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdio.h>

int
main(int argc, char **argv)
{ int f = open("test.data", O_CREAT|O_TRUNC|O_WRONLY, 0666);
  char data[] = "0123456789abcdef";
  char data2[] = "0123456789ABCDEF";
  char zero[16] = {0};

  write(f, data, 16);
  write(f, zero, 16);
  write(f, data, 16);
  lseek(f, 16, SEEK_SET);
  write(f, data2, 16);

  close(f);

  return 0;
}
```

**Comment by rla** [comment]
Related issue: #6381

**Comment by kripken** [comment]
Fixed by #6797.
